### ISSUE TITLE
User defined keyboard shortcuts to access things quicker

### ISSUE THREAD

**Issue by superdreamkilla** [lbalazscs/Pixelitor#134]
Is there a way Pixelitor can have user defined keyboard shortcuts? I use the "starburst" "crystalize" and "four color gradient" options frequent and it would make my workflow much more optimized if I had keyboard shortcuts to quickly access these options.  Is there a way I can compile it from source with defined keyboard shortcuts by editing the source code, or a even a palette search in a future release? The last seems most unlikely but I'm willing to compile a custom pixelitor just to get my favourite filters on quick access. 

Alternatively maybe a way to bookmark frequent filters.

Thanks I really appreciate this great software and the potential it has.

**Comment by lbalazscs** [comment]
1) Compiling with edited source code: should be doable. About building Pixelitor from the sources see the discussion at https://github.com/lbalazscs/Pixelitor/issues/132
Let me know if you have any problems while building Pixelitor.
You would need to edit the following files:
- src/main/java/pixelitor/menus/MenuBar.java
- src/main/java/pixelitor/utils/Keys.java

The changes shouldn't be very difficult, because the files already contain examples. In Keys create a new keystroke by copy-pasting and editing one you like, for example add the line
```java
public static final KeyStroke CTRL_ALT_P = getKeyStroke('P', CTRL + ALT);
```
and in MenuBar replace
```java
sub.addFilter(Starburst.NAME, Starburst::new);
```
with
```java
        sub.buildFilter(Starburst.NAME, Starburst::new)
            .withKey(CTRL_ALT_P)
            .add();
```
and similarly for the other filters.

2) Searching for filters: interesting idea, I will try to add implement it. But this won't be as fast as a keyboard shortcut.
3) I am also thinking about about bookmarking filters, or automatically remembering recent filters (like in recent files in the Files menu)
4) User defined keyboard shortcuts would be nice, but I see no reasonably easy way to implement them. This will probably happen in the foreseeable future only if someone contributes the code.

**Comment by lbalazscs** [comment]
I implemented Filter Search, now it should be easier to start a filter using only the keyboard, and without remembering its exact place in the menus.
![filter_search](https://user-images.githubusercontent.com/9939454/95687558-8f638a80-0c04-11eb-89d7-c33e423d3f76.png)

**Comment by superdreamkilla** [comment]
Okay this is awesome. I am using the 4.2.4 build now.  The filter search seems to be working amazing.

I noticed .pxc files with text layers from 4.2.3 are not preserved.  It's not the end of the world though as I can recreate them.  Can you confirm if this the case on your end too? They still work on 4.23 but not the 4.24 beta.

**Comment by lbalazscs** [comment]
They should still work, but I can confirm, that they don't... I am looking at it now, probably I will fix it soon.

**Comment by lbalazscs** [comment]
The newest GitHub version should be able to open pxc files saved with 4.2.3

**Comment by lbalazscs** [comment]
The implementation is included in the 4.2.4 beta 1: https://github.com/lbalazscs/Pixelitor/releases/tag/v4.2.4-beta1

**Comment by lbalazscs** [comment]
Filter search is included in the released version 4.3.0
